### PR TITLE
Moving to docker

### PR DESCRIPTION
**Pull request by gentlecat**
Also:
* Updated the way initialization works (switched to https://github.com/metabrainz/python-brainzutils)
* Switched from nose to pytest

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM python:3.5.1
+
+RUN apt-get update && apt-get install -y build-essential libffi-dev \
```
Comment (mayhem) [inline Dockerfile]: The best practices document suggests using: --no-install-recommends to keep the size of the resultant images as small as possible.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    build:
+      context: ./docker/nginx_custom/
+      dockerfile: Dockerfile
+    links:
+      - web:web
+    ports:
+      - "80:80"
+
+  web:
+    build: .
```
Comment (mayhem) [inline docker-compose.yml]: Why is the web container not marked as "restart: always"? I've noticed this elsewhere too. Same goes for other docker images...

Comment (gentlecat) [inline docker-compose.yml]: Yeah, I guess there's no reason not to do it.

Comment (gentlecat) [inline docker-compose.yml]: I'll actually split up Compose files into production and development/testing. I don't think restarts are needed in former.

**Comment by gentlecat** [comment]
OK, this one should be good to go now. Just need to test.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    apt-get install -y --no-install-recommends \
+                    build-essential \
+                    libffi-dev \
+                    libssl-dev \
+                    libxml2-dev \
+                    libxslt1-dev
+COPY requirements.txt /code/
+RUN pip install -r requirements.txt
+
+RUN pip install uWSGI==2.0.13.1
```
Comment (mayhem) [inline Dockerfile]: why is this needed? can it live inside requirements.txt ?

File: README.md [hunk 4]
```diff
@@ -11,57 +11,18 @@ into the MusicBrainz pages.
...
-    $ cd sql
-    $ ./setup.sh
-
-### Running
-
-After that app is ready to go.
+To run all tests use:
 
-    $ python mbspotify/server.py
+    $ py.test
```
Comment (mayhem) [inline README.md]: And that tests things running in containers?

File: consul_config.py.ctmpl [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{{- define "KEY" -}}
```
Comment (mayhem) [inline consul_config.py.ctmpl]: Sadly my consul knowledge is non existent still, so I can't really comment.

File: docker/config/MBSpotify/config.prod.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (mayhem) [inline docker/config/MBSpotify/config.prod.json]: what is this used for?

File: docker/uwsgi.ini [hunk 7]
```diff
@@ -0,0 +1,6 @@
+[uwsgi]
+socket = 0.0.0.0:3031
+module = mbspotify
+callable = create_app()
+chdir = /code/
+processes = 20
```
Comment (mayhem) [inline docker/uwsgi.ini]: Is this the non-production version? If so, 20 is really high for non production. Maybe 5?

Comment (gentlecat) [inline docker/uwsgi.ini]: This one is unused. uWSGI is not used for development or running tests.

File: docker/config/MBSpotify/config.prod.json (hunk 6 above)
Comment (gentlecat) [inline docker/config/MBSpotify/config.prod.json]: This is configuration file for git2consul.

File: Dockerfile (hunk 3 above)
Comment (gentlecat) [inline Dockerfile]: It's only need for a production container.

File: README.md (hunk 4 above)
Comment (gentlecat) [inline README.md]: No, that's outside of Docker. Just pushed an update for commands listed there.

**Comment by gentlecat** [comment]
Let me know if I can explain anything.
